services/vehicle/src/test/java/com/wedriveu/services/vehicle/entity/VehicleStoreImplTest.java

```diff
@@ -0,0 +1,265 @@
+        ObjectMapper mapper = new ObjectMapper();
+        List<Vehicle> vehicles = readFromVehiclesDb(mapper);
+        return vehicles;
+    }
+
+    public void updateVehicleInVehicleList(String carLicencePlate, String state, Position position, Date lastUpdate) {
+        ObjectMapper mapper = new ObjectMapper();
+        List<Vehicle> vehicles = readFromVehiclesDb(mapper);
+        for (Vehicle vehicle : vehicles) {
+            if(vehicle.getCarLicencePlate().equals(carLicencePlate)){
+                Log.log("Vehicle found for update");
+                if(!(vehicle.getState().equals(state))) {
+                    vehicle.setState(state);
+                }
+                if(!(vehicle.getPosition().equals(position))) {
+                    vehicle.setPosition(position);
+                }
+                vehicle.setLastUpdate(lastUpdate);
+            }
+        }
+        checkDuplicatesAndWriteOnVehiclesDb(vehicles, mapper);
+    }
+
+    public void deleteVehicleFromDb(String carLicencePlate) {
+        ObjectMapper mapper = new ObjectMapper();
+        List<Vehicle> vehicles = readFromVehiclesDb(mapper);
+        for (Vehicle vehicle : vehicles) {
+            if(vehicle.getCarLicencePlate().equals(carLicencePlate)){
+                Log.log("Vehicle found for delete");
+                vehicles.remove(vehicle);
+                break;
+            }
+        }
+        checkDuplicatesAndWriteOnVehiclesDb(vehicles, mapper);
+    }
+
+    public void replaceVehicleInDb(String carLicencePlateToDelete, Vehicle replacementVehicle) {
+        ObjectMapper mapper = new ObjectMapper();
+        List<Vehicle> vehicles = readFromVehiclesDb(mapper);
+        for (Vehicle vehicle : vehicles) {
+            if(vehicle.getCarLicencePlate().equals(carLicencePlateToDelete)){
+                Log.log("Vehicle found for replacement");
+                vehicles.remove(vehicle);
+                vehicles.add(replacementVehicle);
+                break;
+            }
+        }
+        checkDuplicatesAndWriteOnVehiclesDb (vehicles, mapper);
+    }
+
+    public boolean theVehicleIsInTheDb(String carLicensePlate){
+        List<Vehicle> vehicles = getVehicleList();
+        for (Vehicle vehicle : vehicles) {
+            if(vehicle.getCarLicencePlate().equals(carLicensePlate)){
+                Log.log("Vehicle found");
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean thereAreNotDuplicates(List<Vehicle> vehicles) {
+        for (int j = 0; j < vehicles.size(); j++) {
+            String currentCarLicencePlate = vehicles.get(j).getCarLicencePlate();
+            for (int i = 0; i < vehicles.size(); i++) {
+                if (j != i
+                        && i < vehicles.size() - 1
+                        && vehicles.get(i).getCarLicencePlate().equals(currentCarLicencePlate)) {
+                    Log.log("There are two duplicates car licence plate: "
```

Comment:
same question as above. Logs are good to understand what happens, but in production environment they're useless and make the method less easily readable. Are there really needed here?